### PR TITLE
Fix account creation case in eth_sendTransaction

### PR DESCRIPTION
**Pull request by gtrepta**
fixes: https://github.com/runtimeverification/firefly/issues/241

### CODE REVIEW

File: web3.md [hunk 1]
```diff
@@ -852,7 +852,7 @@ Transaction Receipts
          </message>
          <account>
            <acctID> ACCTFROM </acctID>
-           <balance> BAL => BAL -Int (GLIMIT *Int GPRICE) </balance>
+           <balance> BAL => BAL -Int 2 *Int (GLIMIT *Int GPRICE) </balance>
```
Comment (gtrepta) [inline web3.md]: For some reason, the `from` account ends with too much gas being refunded (as if it is being refunded twice), and so far I haven't been able to find out where this is happening.

Comment (ehildenb) [reply web3.md]: @gtrepta it looks like `finalizeTx` is already refunding gas to the account (at the same time that it awards the miner gas.

So you probably don't need `#catchHaltTx _` to do a `#refund GAVAIL`

File: web3.md [hunk 2]
```diff
@@ -891,9 +886,21 @@ Transaction Receipts
...
- // -------------------------------
-    rule <k> #halt ~> #catchHaltTx => . ... </k>
+    syntax KItem ::= "#catchHaltTx" Account
+ // ---------------------------------------
+    rule <statusCode> _:ExceptionalStatusCode </statusCode>
+         <k> #halt ~> #catchHaltTx _ => #popCallStack ~> #popWorldState ... </k>
+
+    rule <statusCode> EVMC_REVERT </statusCode>
+         <k> #halt ~> #catchHaltTx _ => #popCallStack ~> #popWorldState ~> #refund GAVAIL ... </k>
+         <gas> GAVAIL </gas>
```
Comment (ehildenb) [inline web3.md]: Here you probably don't need to do the `#refund GAVAIL`

**Comment by ehildenb** [comment]
You'll have to look carefully at the refunds that `finalizeTx` and `mkCodeDeposit` make. it looks like `finalizeTx` already awards a gas refund, and so does `mkCodeDeposit` when you have `EVMC_SUCCESS` by calling `finalizeCodeDeposit`.

File: web3.md (hunk 1 above)
Comment (gtrepta) [reply web3.md]: I don't think that's it. `#refund` adds to the `<gas>` cell. It's used after a `#popCallStack` to preserve the remaining gas.

**Pull request merged**